src/midend/atomics_op_lower.cpp

```diff
@@ -141,13 +160,31 @@ bool graphit::AtomicsOpLower::ApplyExprVisitor::lowerCompareAndSwap(mir::FuncExp
                 FieldVectorProperty field_vector_prop = tensor_array_read_expr->field_vector_prop_;
                 mir::Type::Ptr field_type = mir_context_->getVectorItemType(field_name);
 
+                //TODO this is hack to access local array type since it is not registered in the context map
+                //it has to be scalar tensor read to have lower to atomic operations
+                if (! mir::isa<mir::VarExpr>(tensor_array_read_expr->target)){
```

Comment:
Use an assert here? 
Print some error message if this assumption is not true.

Reply:
Will do! Thanks

Reply:
actually this method (lowerCompareAndSwap) returns bool instead of void even though the return value is never used. Should we modify it to void?